### PR TITLE
Adding APILifeCycle.xml and related executor+dependencies

### PR DESCRIPTION
**Pull request by thushara35**

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/lifecycles/APILifeCycle.xml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            <scxml xmlns="http://www.w3.org/2005/07/scxml"
+                   version="1.0"
+                   initialstate="Created">
+
+                <state id="Created">
+                    <datamodel>
+                        <data name="transitionExecution">
+                            <execution forEvent="Promote"
+                                       class="org.wso2.carbon.apimgt.impl.executors.ApiPublisherExecutor">
+                                <parameter name="apim.test" value="Dummy"/>
```
Comment (lalaji) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/lifecycles/APILifeCycle.xml]: better add a comment,why add the dummy parameter

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/executors/ApiPublisherExecutor.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            GenericArtifact apiArtifact = artifactManager.getGenericArtifact(artifactId);
+            API api = APIUtil.getAPI(apiArtifact);
+            APIStatus newStatus = getApiStatus(targetState);
+            APIProvider apiProvider = APIManagerFactory.getInstance().getAPIProvider(user);
+            failedGateways = apiProvider.changeAPIStatus(api, newStatus, user, true);
+            if (failedGateways == null || failedGateways.size() == 0) {
+                executed = true;
+            }
+        } catch (RegistryException e) {
+            log.error("Failed to get the generic artifact ", e);
```
Comment (lalaji) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/executors/ApiPublisherExecutor.java]: change the exception messages

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/executors/ApiPublisherExecutor.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This class is an implementation of the interface {@link org.wso2.carbon.governance.registry.extensions.interfaces.Execution}
+ * This class consists methods that will publish an API to API Manager.
+ *
+ * This class gets initiated when a REST Service lifecycle is added to a REST Service. In initialization following
+ * static configuration parameters should be defined in the lifecycle.
```
Comment (lalaji) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/executors/ApiPublisherExecutor.java]: Class comment need to be changed a bit.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/lifecycles/APILifeCycle.xml (hunk 1 above)
Comment (thushara35) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/lifecycles/APILifeCycle.xml]: SameeraM has done the fix, we can remove the < parameter >

**Pull request merged**